### PR TITLE
Add functionality to run Publishing e2e tests

### PR DESCRIPTION
**Pull request by emmabeynon**
https://trello.com/c/NwvQLJMs/957-run-e2e-tests-as-part-of-content-store-and-travel-advice-publisher-builds-1

### CODE REVIEW

File: modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy [hunk 1]
```diff
@@ -746,4 +757,22 @@ def setBuildStatus(repoName, commit, message, state) {
   ]);
 }
 
+def runPublishingE2ETests() {
+  build(
+    job: "publishing-e2e-tests/${params.PUBLISHING_E2E_TESTS_BRANCH}",
+    parameters: [
+      [$class: "StringParameterValue",
+        name: "PUBLISHING_API_COMMITISH",
```
Comment (kevindew) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: I think this needs to be just the application that is being tested.

E.g. CONTENT_STORE_COMMITISH in this instance. Which is something e2e tests would need to support

Comment (emmabeynon) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: I've updated this to take the app name, but would be good to chat about how best to implement this in the e2e tests Jenkinsfile.

File: modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy [hunk 2]
```diff
@@ -188,6 +190,19 @@ def buildProject(Map options = [:]) {
...
 
+    if (options.publishingE2eTests == true) {
+      setEnvar("FULL_COMMIT_HASH", sh(
+        script: "git rev-parse HEAD",
+        returnStdout: true
+      )
+      stage("End-to-end tests") {
+        appCommitishName = env.PUBLISHING_E2E_TESTS_APP_PARAM ||
+          repoName.replace("-", "_").toUpperCase() + "_COMMITISH"
+        testBranch = env.PUBLISHING_E2E_TESTS_BRANCH || "master"
```
Comment (kevindew) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: we've [very recently](https://github.com/alphagov/publishing-api/commit/835956e8fc22a23723b4814d8b693a82f4580c76) switched to using "test-against" as the default branch

**Comment by emmabeynon** [comment]
> I'm not hugely confident with the groovy - have you been able to verify it works? we might want to deploy a branch of puppet first to check it if you're at all unsure.

Not yet, that's what I'll do next 👍

File: modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy (hunk 2 above)
Comment (emmabeynon) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: Updated :)

File: modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy [hunk 3]
```diff
@@ -188,6 +190,26 @@ def buildProject(Map options = [:]) {
       }
     }
 
+    if (options.publishingE2eTests == true) {
+      setEnvar("FULL_COMMIT_HASH", sh(
+        script: "git rev-parse HEAD",
+        returnStdout: true
+      ))
+      stage("End-to-end tests") {
+        if ( env.PUBLISHING_E2E_TESTS_APP_PARAM == null ) {
```
Comment (emmabeynon) [inline modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy]: @kevindew I've decided to use an if/else statement to set `appCommitishName` and `testBranch` as I found using a ternary statement exceeded the line length too much.

**Comment by emmabeynon** [comment]
I've tested this in integration and publishing end-to-end tests are running successfully for content-store 🎉

**Pull request merged**